### PR TITLE
The compiler path selection control is not in sync with the textbox

### LINKED ISSUE #7427: The compilerPath option does not match the tick in front of the option in the drop-down box (Mac OS only)
Bug type: Language Service

**Describe the bug**
- OS: Mac and Mac ARM64
- C/C++ Extension Version: 1.3.0
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. Open IntelliSense Configuration UI
2. Change the compiler path selection option.
3. The option does not match the tick in front of the option in the drop-down box (select any of the compiler paths, re-open the drop-down box, the tick is in the first path unchanged)
image

**Expected behavior**
The "check mark" should be in front of the corresponding compilation path.

### PR DESCRIPTION
**Pull request by bobbrow**
Fixes: #7427

The faux editable combobox we use to set the compilerPath property was not keeping the select control in sync with the contents of the textbox.  This change fixes it.

### CODE REVIEW

**Comment by bobbrow** [comment]
Suddenly it doesn't compile. I'm investigating why. It has to do with importing `path.normalize` which didn't repro when I was testing, but now it does. :(

File: Extension/ui/settings.ts [hunk 1]
```diff
@@ -212,10 +212,24 @@ class SettingsApp {
...
+    }
 
-        // Reset selection to none
-        el.value = "";
+    // To enable custom entries, the compiler path control is a text box on top of a select control.
+    // This function ensures that the select control is updated when the text box is changed.
+    private fixKnownCompilerSelection(): void {
+        const compilerPath = <HTMLInputElement>document.getElementById(elementId.compilerPath);
+        const knownCompilers = <HTMLSelectElement>document.getElementById(elementId.knownCompilers);
+        for (let n = 0; n < knownCompilers.options.length; n++) {
```
Comment (sean-mcmanus) [inline Extension/ui/settings.ts:222]: Can you do 

```typescript
        for (let n = 0; n < knownCompilers.options.length; n++) {
            if (compilerPath.value.toLowerCase() === knownCompilers.options[n].value.toLowerCase()) {
                knownCompilers.value = knownCompilers.options[n].value;
                return;
            }
            knownCompilers.value = '';
        }
````

File: Extension/ui/settings.ts [hunk 2]
```diff
@@ -212,10 +212,24 @@ class SettingsApp {
...
             command: "knownCompilerSelect"
         });
+    }
 
-        // Reset selection to none
-        el.value = "";
+    // To enable custom entries, the compiler path control is a text box on top of a select control.
+    // This function ensures that the select control is updated when the text box is changed.
+    private fixKnownCompilerSelection(): void {
+        const compilerPath = <HTMLInputElement>document.getElementById(elementId.compilerPath);
```
Comment (sean-mcmanus) [inline Extension/ui/settings.ts]: Can you do something like

```typescript
const compilerPath = (<HTMLInputElement>document.getElementById(elementId.compilerPath)).value.toLowerCase();
```

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

File: Extension/ui/settings.ts (hunk 1 above)
Comment (bobbrow) [reply Extension/ui/settings.ts:222]: I refactored this into a function after some testing revealed I needed to do it in two places, but I didn't take advantage of the fact that I could do an early return now.

I don't think I want to set the value to `''` N times during this loop though.  What I should do is move that out of the loop and let the early return handle the other case.

File: Extension/ui/settings.ts (hunk 2 above)
Comment (bobbrow) [reply Extension/ui/settings.ts]: yes, I can avoid lowercasing this N times.  Good catch.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**